testing-common/src/main/groovy/io/opentelemetry/instrumentation/test/InMemoryExporter.java

```diff
@@ -290,37 +214,8 @@ private long getMinSpanOrder(List<SpanData> spans) {
     return orderedSpans;
   }
 
-  private void sortOneLevel(List<Node> nodes) {
-    Collections.sort(
-        nodes,
-        new Comparator<Node>() {
-          @Override
-          public int compare(Node node1, Node node2) {
-            return Ints.compare(getSpanOrder(node1.span), getSpanOrder(node2.span));
-          }
-        });
-  }
-
-  private int getSpanOrder(SpanData span) {
-    Integer order = spanOrders.get(span.getSpanId());
-    if (order == null) {
-      throw new IllegalStateException("order not found for span: " + span);
-    }
-    return order;
-  }
-
-  // trace is completed if root span is present
```

Comment:
This statement is wrong I think - async instrumentation would have root spans completing early. I'm not sure why this seemed to work ok before my change, but I couldn't get most async tests to pass until changing the waiting logic to be closer to what we do in smoke tests.